rust/connlib/tunnel/src/client.rs

```diff
@@ -306,11 +320,51 @@ impl ClientState {
             next_dns_refresh: Default::default(),
             node: ClientNode::new(private_key),
             system_resolvers: Default::default(),
+            sites_status: Default::default(),
+            gateways_site: Default::default(),
         }
     }
 
-    fn resources(&self) -> Vec<ResourceDescription> {
-        self.resource_ids.values().sorted().cloned().collect_vec()
+    pub(crate) fn resources(&self) -> Vec<callbacks::ResourceDescription> {
```

Comment:
I think this creates a weird coupling to the callbacks. What if we just make this a tuple for now instead of duplicating the type?

Long-term, I'd like connlib to have a proper domain model and not (ab)use the DTO types with the portal. The new `ResourceDescription` is yet another DTO and not a proper domain model.

Reply:
the `resources` function is only used after all for the callbacks.

We could also return a tuple here and create the `callbacks::ResourceDescription` on the top function but that just seems like a bit more convoluted that simply returning the `callbacks::ResourceDescription` here.